### ISSUE TITLE
Mouse position for menu item selection offset in Chrome and Firefox

### ISSUE THREAD

**Issue by maehler** [cytoscape/cytoscape.js-cxtmenu#75]
If the cytoscape container is offset by scrolling, the mouse position for the menu item selection is offset as well. If the page is not scrolled then the selection works as expected. I see this behaviour in Chrome 70.0.3538.110 and Firefox 62.0.3, but it behaves as expected in Safari 12.0.1. Here is a jsfiddle demonstrating the issue (assuming that the browser window is small enough): https://jsfiddle.net/3rfw1qja/7/. Triggering the context menu on the top node should work fine, but scrolling down to the bottom one and triggering the menu should demonstrate the issue.

**Comment by maehler** [comment]
I'm on Mac 10.14.1, updated my browsers and disabled all plugins, but the issue remains.

**Comment by dilema** [comment]
Same problem here on Mac 10.12.6.
Firefox 65.0.1 / Chrome 72.0.3626.119 have this issue for me but it works fine in Safari 12.0.3.
And it looks like the more you scroll, the more offset it is. I changed the cy css height property to `height: 10000px`, and when triggering the context menu of the bottom node, the pointer barely even move.

**Comment by maxkfranz** [comment]
This is an odd issue.  I can't reproduce it outside of JSFiddle.

This is your code pasted into the demo page:

```html
<!DOCTYPE>

<html>

	<head>
		<title>cytoscape-cxtmenu.js demo</title>

		<meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1, maximum-scale=1">

		<link href="https://unpkg.com/font-awesome@4.7.0/css/font-awesome.min.css" rel="stylesheet" type="text/css" />
		<script src="https://unpkg.com/cytoscape/dist/cytoscape.min.js"></script>

		<!-- for testing with local version of cytoscape.js -->
		<!--<script src="../cytoscape.js/build/cytoscape.js"></script>-->

		<script src="cytoscape-cxtmenu.js"></script>

		<style>
			body {
				font-family: helvetica;
				font-size: 14px;
			}

			#cy {
				background-color: red;
				height: 1000px;
				width: 50%;
			}

		</style>

		<script>
			window.addEventListener('DOMContentLoaded', function(){

				var cy = window.cy = cytoscape({
					container: document.getElementById('cy'),

					elements: [
						{
							group: 'nodes',
							data: {
								id: 'n1'
							}
						},
						{
							group: 'nodes',
							data: {
								id: 'n2'
							}
						},
						{
							group: 'edges',
							data: {
								source: 'n1',
								target: 'n2'
							}
						}
					]
				});

				var menu = cy.cxtmenu({
					commands: [{
						content: 'testing',
						select: () => console.log('testing')
					}, {
						content: 'testing2',
						select: () => console.log('testing2')
					}, {
						content: 'testing3',
						select: () => console.log('testing3')
					}]
				});

			});
		</script>
	</head>

	<body>
		<div id="cy"></div>

	</body>

</html>

```

You can replace the demo page with this one to test.  See the screen capture:
![beam](https://user-images.githubusercontent.com/989043/54056514-4a70aa00-41be-11e9-9f73-b6e27047c20f.gif)

Can you reproduce the issue with this file or a modified version of this file?

**Comment by maehler** [comment]
Just like you, I cannot reproduce the issue using the demo page. I've tried to modify it with different nested element structures, and having other elements push down the network container, but it still works as it should. I'm starting to suspect that the issue is due to some element/style combinations in the project where I'm seeing this issue, as well as in the JSFiddle. If I ever manage to pinpoint the cause, I will let you know here.

**Comment by maehler** [comment]
I think I found what triggers this now. The doctype of your example is `<!DOCTYPE>` (so quirks mode I assume), but changing this to `<!DOCTYPE html>` results in the offset that we're seeing in the JSFiddle. This is probably not the ultimate cause of the issue, but it might help in tracking it down.

**Comment by randresf** [comment]
Hello, when can we have this fix merged? I just installed the pkg but the previous commit was not there and I am having this issue. Thanks

**Comment by maxkfranz** [comment]
I used this page for testing, based on @maehler's suggestion:

```html
<!DOCTYPE html>

<html>

	<head>
		<title>cytoscape-cxtmenu.js demo</title>

		<meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1, maximum-scale=1">

		<link href="https://unpkg.com/font-awesome@4.7.0/css/font-awesome.min.css" rel="stylesheet" type="text/css" />
		<script src="https://unpkg.com/cytoscape/dist/cytoscape.min.js"></script>

		<!-- for testing with local version of cytoscape.js -->
		<!--<script src="../cytoscape.js/build/cytoscape.js"></script>-->

		<script src="cytoscape-cxtmenu.js"></script>

		<style>
			body {
				font-family: helvetica;
				font-size: 14px;
			}

			#cy {
				background-color: red;
				height: 1000px;
				width: 800px;
			}

		</style>

		<script>
			window.addEventListener('DOMContentLoaded', function(){

				var cy = window.cy = cytoscape({
					container: document.getElementById('cy'),

					elements: [
						{
							group: 'nodes',
							data: {
								id: 'n1'
							}
						},
						{
							group: 'nodes',
							data: {
								id: 'n2'
							}
						},
						{
							group: 'edges',
							data: {
								source: 'n1',
								target: 'n2'
							}
						}
					]
				});

				var menu = cy.cxtmenu({
					commands: [{
						content: 'testing',
						select: () => console.log('testing')
					}, {
						content: 'testing2',
						select: () => console.log('testing2')
					}, {
						content: 'testing3',
						select: () => console.log('testing3')
					}]
				});

			});
		</script>
	</head>

	<body>
		<div id="cy"></div>

	</body>

</html>
```

You can use the arrow keys to scroll the page both vertically and horizontally to test.  Note that @deelef's commit only addresses the vertical scroll.  To make contributions to any of the Cytoscape.js packages, please make pull requests in future.

**Comment by maxkfranz** [comment]
Included in v3.1.1 : https://www.npmjs.com/package/cytoscape-cxtmenu/v/3.1.1
